File: backend/schemas/calendar.py
from datetime import date

from pydantic import BaseModel


class CalendarQuery(BaseModel):
    start_date: date
    end_date: date


class CalendarEvent(BaseModel):
    date: date
    instructor_id: str
    instructor_name: str
    course_id: str
    course_title: str
    course_status: str | None = None
    assignment_status: str | None = None
    class_name: str | None = None
    assignment_id: str
    notion_page_id: str = ""
    workbook_full_url: str | None = None
    manager: str | None = None
    manager_email: str | None = None
    sales_rep: str | None = None
    sales_rep_email: str | None = None


class CalendarResponse(BaseModel):
    events: list[CalendarEvent]

File: frontend/src/api/calendar.ts
import { fetchWithAuth, handleResponse } from './client'

export interface CalendarEvent {
  date: string
  instructor_id: string
  instructor_name: string
  course_id: string
  course_title: string
  course_status: string | null
  assignment_status: string | null
  class_name: string | null
  assignment_id: string
  notion_page_id: string
  workbook_full_url: string | null
  manager: string | null
  sales_rep: string | null
}

export interface CalendarResponse {
  events: CalendarEvent[]
}

export async function getCalendar(startDate: string, endDate: string): Promise<CalendarResponse> {
  const params = new URLSearchParams({ start_date: startDate, end_date: endDate })
  const res = await fetchWithAuth(`/api/calendar?${params}`)
  return handleResponse<CalendarResponse>(res)
}
